### PR TITLE
ElasticSearch and JavaScript State Machine updates

### PR DESCRIPTION
**Pull request by jppellerin**
case 2. Improvement to existing type definition.
- Added Create to ElasticSearch
  - https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html#api-reference
- Added Transition to JavaScript-State-Machine
  - https://github.com/jakesgordon/javascript-state-machine

### CODE REVIEW

File: state-machine/state-machine.d.ts [hunk 1]
```diff
@@ -66,6 +66,10 @@ interface StateMachineCan {
 	(evt: string): boolean;
 }
 
+interface IStateMachineTransitions {
```
Comment (vvakame) [inline state-machine/state-machine.d.ts]: remove `I` prefix.

**Comment by vvakame** [comment]
cc/ @CasperSkydt

File: elasticsearch/elasticsearch.d.ts [hunk 2]
```diff
@@ -102,6 +104,21 @@ declare module Elasticsearch {
         index?: string;
     }
 
+    export interface ClientCreateParams extends GenericParams {
```
Comment (CasperSkydt) [inline elasticsearch/elasticsearch.d.ts]: rename interface to CreateDocumentParams. Its more similar to the other document operation input parameters.

**Comment by jppellerin** [comment]
@CasperSkydt, @vvakame Updated based on feedback.

**Pull request merged**